compiler-plugin/src/main/kotlin/com/google/devtools/ksp/Incremental.kt

```diff
@@ -446,8 +454,11 @@ class IncrementalContext(
 
     // TODO: Don't do anything if processing failed.
     private fun updateCaches(dirtyFiles: Collection<File>, outputs: Set<File>, sourceToOutputs: Map<File, Set<File>>) {
-        updateSourceToOutputs(dirtyFiles, outputs, sourceToOutputs)
-        updateLookupCache(dirtyFiles)
+        // dirtyFiles may contain new files, which are unknown to sourceToOutputsMap.
+        val oldOutputs = dirtyFiles.flatMap { sourceToOutputsMap[it] ?: emptyList() }
```

Comment:
Is a `distinct()` needed in case an output file correspond to multiple sources?

Reply:
Done. Good catch!